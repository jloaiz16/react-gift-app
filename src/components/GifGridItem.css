.gif-container {
  border-radius: 5px;
  border: 1px solid #ccc;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.gif-container img {
  width: 90%;
}
