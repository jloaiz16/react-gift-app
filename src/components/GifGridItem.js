import React from 'react';
import PropTypes from 'prop-types';
import './GifGridItem.css';

const GifGridItem = ({ item }) => {
  return (
    <div className='gif-container animate__bounceIn'>
      <h4> {item.title}</h4>
      <img src={item.url} alt={item.title} />
    </div>
  );
};

GifGridItem.propTypes = {
  item: PropTypes.object.isRequired,
};

export default GifGridItem;
