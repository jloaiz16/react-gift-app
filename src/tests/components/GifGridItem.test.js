import { shallow } from 'enzyme';
import GifGridItem from '../../../src/components/GifGridItem';

describe('<GifGridItem /> suit test', () => {
  const title = 'Test';
  const url = 'https://media.giphy.com/media/X7s4uckfyihGJDrSpo/giphy.gif';
  const wrapper = shallow(
    <GifGridItem
      item={{
        id: '123ASDFA',
        title: title,
        url: url,
      }}
    />
  );

  test('Should render <GifGridItem /> properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should render <h4> with the title', () => {
    expect(wrapper.find('h4').text().trim()).toBe(title);
  });

  test('Should render <img> with the url and title', () => {
    const imgElement = wrapper.find('img').getElement();
    expect(imgElement.props.src).toBe(url);
    expect(imgElement.props.alt).toBe(title);
  });

  test('Should have class animate__bounceIn', () => {
    const divElement = wrapper.find('div').getElement();
    expect(
      divElement.props.className
        .split(' ')
        .some((item) => item === 'animate__bounceIn')
    ).toBe(true);
  });
});
