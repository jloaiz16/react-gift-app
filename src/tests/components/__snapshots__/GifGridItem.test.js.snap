// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> suit test Should render <GifGridItem /> properly 1`] = `
<div
  className="gif-container animate__bounceIn"
>
  <h4>
     
    Test
  </h4>
  <img
    alt="Test"
    src="https://media.giphy.com/media/X7s4uckfyihGJDrSpo/giphy.gif"
  />
</div>
`;
